### PR TITLE
#33: Add edit button which redirects to GitHub

### PR DESCRIPTION
**Pull request by marcinwrobel1986**
### THIS PR:
- added `edit_button_handler.py` script for adding `edit` icon to generated html pages
- added step to `deploy_docs.yml` (GitHub action)
- added `edit-button` section to make file in order to simplify edit button generation locally

@dalg24 @fnrizzi Please take a look. 

Documentation DEMO with working button on the bottom right could be found [here](https://marcinwrobel1986.github.io/kokkos-core-wiki/)

Below are sample screenshots (2nd with mouse over edit button):
Zrzut ekranu z 2022-07-15 10-20-48
Zrzut ekranu z 2022-07-15 10-21-02

### CODE REVIEW

**Comment by fnrizzi** [comment]
@marcinwrobel1986 this look good, i would have these suggestions though: 
- can you please show how the button would look like if placed at the top right of the page instead of bottom? 
- can you please change the button link to take to the source file 
- can you please change he text "edit on github" to "take me to souce file" 

I think we need to avoid editing files directly on github

**Comment by marcinwrobel1986** [comment]
> @marcinwrobel1986 this look good, i would have these suggestions though:
> 
> * can you please show how the button would look like if placed at the top right of the page instead of bottom?

IMHO the place on the bottom right is the only one which will not interfered with the theme. Current position is the same for every page

> * can you please change the button link to take to the source file

Done.

> * can you please change he text "edit on github" to "take me to source file" ? I think we need to avoid editing files directly on github.

Done as well.

**Comment by dalg24** [comment]
I pulled your branch and built it locally and cannot see the button.

**Comment by dalg24** [comment]
I also would slightly prefer the `Edit on GitHub` to be in the header, just like with the ReadTheDocs theme.

**Comment by dalg24** [comment]
> you need to run Python script python3 source/edit_button_handler.py

We need to add to the Makefile too

**Comment by marcinwrobel1986** [comment]
> > you need to run Python script python3 source/edit_button_handler.py
> 
> ~We need to add to the Makefile too~ Actually I would say it needs to be added to the html rule in the Makefile and nowhere else

Hey @dalg24 Damien,
I implemented this more elegant version with edit button at the top, also Makefile is modified and used in CI.

**Review by fnrizzi — APPROVED** [review]

**Comment by fnrizzi** [comment]
i think this looks nice ! maybe to make it more visible we could colorcode it? just a nitpick comment

File: docs/Makefile [hunk 1]
```diff
@@ -3,3 +3,6 @@ html:
 
 clean:
 	rm -rf ./generated_docs/
+
+edit-button:
+	python3 ./source/edit_button_handler.py
```
Comment (dalg24) [inline docs/Makefile:8]:
```suggestion
	python3 ./source/edit_button_handler.py

all:
	make html
	make edit-button
```
Or 
```
html:
	sphinx-build -b html ./source/ ./generated_docs/
	python3 ./source/edit_button_handler.py

Comment (marcinwrobel1986) [reply docs/Makefile]: @dalg24 done

**Review by dalg24 — APPROVED** [review]

**Pull request merged**